### DISCUSSION TITLE
Form isn't resetting to use defaultValues when unmounted

### DISCUSSION THREAD

**Discussion by jhsu** [react-hook-form/react-hook-form#2940] [Q&A] [answered]
When using `shouldUnregister: false` and using the `Controller` for some custom components, the fields do not reset to use defaultValues. I think this has something to do with the defaultValues object being mutated somewhere.

see the scodesandbox: https://codesandbox.io/s/react-hook-form-get-started-forked-2tdli?file=/src/index.js

something seems to modify the defaultValues object itself (you can see the logged `defaultData`)

A workaround for this is to not use a default values object outside the component (and not use a prop) and construct it in the component's body.

**Comment by bluebill1049** [comment, marked as answer]
Thanks for the demo, make my life a bit easier, what're the steps to reproduce this issue? happy to investigate further on this.

**Reply by jhsu** [reply]
Hey, to reproduce the issue, start typing in the "Controller field" textarea, then press the "toggle" button at the top. Then press "toggle" once more to bring back the form. you can see that the "controller field" persisted the previous value entered.

**Reply by bluebill1049** [reply]
Thanks @jhsu 👍

**Reply by bluebill1049** [reply]
here is the issue: https://github.com/react-hook-form/react-hook-form/issues/2955

**Reply by bluebill1049** [reply]
patched.

**Reply by jhsu** [reply]
Thanks!
